nova_vm/src/ecmascript/builtins/reflection/reflect_object.rs

```diff
@@ -131,92 +141,272 @@ impl Builtin for ReflectObjectSetPrototypeOf {
+        // 2. Let key be ? ToPropertyKey(propertyKey).
+        let key = to_property_key(agent, arguments.get(1))?;
+        // 3. Let desc be ? target.[[GetOwnProperty]](key).
+        let desc = target.internal_get_own_property(agent, key)?;
+        // 4. Return FromPropertyDescriptor(desc).
+        match PropertyDescriptor::from_property_descriptor(desc, agent) {
+            Some(ret) => Ok(ret.into_value()),
+            None => Ok(Value::Undefined),
+        }
     }
 
     fn get_prototype_of(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!();
+        // 1. If target is not an Object, throw a TypeError exception.
+        if !arguments.get(0).is_object() {
+            return Err(agent.throw_exception(ExceptionType::TypeError, "Value is not an object"));
+        }
+        let target = Object::try_from(arguments.get(0)).unwrap();
+        // 2. Return ? target.[[GetPrototypeOf]]().
+        match target.internal_get_prototype_of(agent)? {
+            Some(ret) => Ok(ret.into_value()),
+            None => Ok(Value::Null),
+        }
     }
 
-    fn has(_agent: &mut Agent, _this_value: Value, _arguments: ArgumentsList) -> JsResult<Value> {
-        todo!();
+    fn has(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
+        // 1. If target is not an Object, throw a TypeError exception.
+        if !arguments.get(0).is_object() {
+            return Err(agent.throw_exception(ExceptionType::TypeError, "Value is not an object"));
+        }
+        let target = Object::try_from(arguments.get(0)).unwrap();
+
+        // 2. Let key be ? ToPropertyKey(propertyKey).
+        let key = to_property_key(agent, arguments.get(1))?;
+        // 3. Return ? target.[[HasProperty]](key).
+        let ret = target.internal_has_property(agent, key)?;
+        Ok(ret.into())
     }
 
     fn is_extensible(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!();
+        // 1. If target is not an Object, throw a TypeError exception.
+        if !arguments.get(0).is_object() {
+            return Err(agent.throw_exception(ExceptionType::TypeError, "Value is not an object"));
+        }
+        let target = Object::try_from(arguments.get(0)).unwrap();
+        // 2. Return ? target.[[IsExtensible]]().
+        let ret = target.internal_is_extensible(agent)?;
+        Ok(ret.into())
     }
 
     fn own_keys(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!();
+        // 1. If target is not an Object, throw a TypeError exception.
+        if !arguments.get(0).is_object() {
```

Comment:
**nitpick:** We've usually used `let Ok(target) = Object::try_from(object) else {` in these sorts of places.